pkg/api/types.go

```diff
@@ -1414,8 +1412,13 @@ type Toleration struct {
 	// Effect indicates the taint effect to match. Empty means match all taint effects.
 	// When specified, allowed values are NoSchedule and PreferNoSchedule.
 	Effect TaintEffect `json:"effect,omitempty"`
-	// TODO: For forgiveness (#1574), we'd eventually add at least a grace period
-	// here, and possibly an occurrence threshold and period.
+	// ForgivenessSeconds Represents the period of time the toleration tolerates the taint.
```

Comment:
And worse yet, the heritage is inconsistent with this new definition!

So yes, any of these is better than ForgivenessSeconds, but the free-standing term "Period" seems to implies a notion of repetition to me.  How about durationSeconds?